### PR TITLE
Sprint12 04 a45619 lb manual review

### PR DESCRIPTION
**Pull request by mrusan**

### CODE REVIEW

File: cookbooks/lb/metadata.rb [hunk 1]
```diff
@@ -23,118 +23,79 @@
...
-  :recipes => [
-                'lb::default',
-                'lb::do_attach_request',
-                'lb::handle_attach',
-                'lb::do_detach_request',
-                'lb::handle_detach',
-                'lb::setup_load_balancer',
-                'lb::do_attach_all'
-                ]
+  :recipes => ["lb::default", "lb::do_attach_request", "lb::handle_attach", "lb::do_detach_request", "lb::handle_detach", "lb::setup_load_balancer", "lb::do_attach_all"]
```
Comment (douglaswth) [inline cookbooks/lb/metadata.rb]: It is actually helpful in this case to have the recipes listed on their own lines. It should be reformatted to something like:

``` ruby
  :recipes => [
    "lb::default",
    ...
  ]
```

You should do this with the other `:recipes` lists too.

File: cookbooks/lb/recipes/default.rb [hunk 2]
```diff
@@ -13,22 +13,20 @@ class Chef::Recipe
...
+    persist true # Store this resource in node between converges.
     action :nothing
   end
 end
 
 gem_package "right_aws" do
   gem_binary "/opt/rightscale/sandbox/bin/gem"
-  action :nothing
-end.run_action(:install)
+end
```
Comment (douglaswth) [inline cookbooks/lb/recipes/default.rb:28]: This should not be removed. The installation of the gem needs to be done at compile time rather than converge time since other recipes actually use the gem. Please reformat it in the following way:

``` ruby
r = gem_package "right_aws" do
  ...
  action :nothing
end
r.run_action(:install)
```

File: cookbooks/lb/recipes/do_attach_all.rb [hunk 3]
```diff
@@ -26,16 +26,11 @@ class Chef::Recipe
   deployment_servers = query_appservers(vhost_name)
 
   # Send warning if no application servers are found.
-  log "  No application servers found" do
-    only_if { deployment_servers.empty? }
-    level :warn
-  end
+  log "  No application servers found" unless deployment_servers
```
Comment (douglaswth) [inline cookbooks/lb/recipes/do_attach_all.rb]: Please revert this. You removed the `level :warn` and it is better to have the conditional logic in the converge phase.

File: cookbooks/lb/recipes/do_attach_all.rb [hunk 4]
```diff
@@ -26,16 +26,11 @@ class Chef::Recipe
...
 
   # Add any servers in deployment not in config.
   servers_to_attach = Set.new(deployment_servers.keys) - inconfig_servers
-  log "  No servers to attach" do
-    only_if { servers_to_attach.empty? }
-  end
+  log "  No servers to attach" unless servers_to_attach
```
Comment (douglaswth) [inline cookbooks/lb/recipes/do_attach_all.rb]: Please revert this as well.

File: cookbooks/lb/recipes/do_attach_all.rb [hunk 5]
```diff
@@ -74,16 +69,14 @@ class Chef::Recipe
...
       app_servers_detached += 1
     else
       log "  Threshold not reached for #{uuid} : #{node[:lb][:threshold][vhost_name][uuid]}"
     end
   end
 
-  log "  No servers to detach" do
-    only_if { app_servers_detached == 0 }
-  end
+  log "  No servers to detach" if app_servers_detached == 0
```
Comment (douglaswth) [inline cookbooks/lb/recipes/do_attach_all.rb]: Same as before, please revert this.

File: cookbooks/lb/libraries/helper.rb [hunk 6]
```diff
@@ -17,31 +20,35 @@ def get_attached_servers(vhost_name)
...
+          port = RightScale::Utils::Helper.get_tag_value('appserver:listen_port', tags)
+          app_servers[uuid] = {}
           app_servers[uuid][:ip] = ip
           app_servers[uuid][:backend_port] = port.to_i
         end
 
-        return app_servers
-      end
+        app_servers
+      end # def query_appservers(vhost_name)
```
Comment (ekargard) [inline cookbooks/lb/libraries/helper.rb:51]: These lines are functionally equivalent.  I personally like the explicit return - it is easier to understand.  I know what return does.  Without it I have to think that oh yeah since it is the last line before the end of the block it is what is going to be returned.

File: cookbooks/lb/metadata.rb [hunk 7]
```diff
@@ -23,118 +23,79 @@
...
   :display_name => "Virtual Host Names",
-  :description => "Comma-seperated list of host names for which the load balancer will answer website requests. First entry will be the default backend and will answer for all host names not listed here.  A single entry of any name, e.g. 'default' or 'applistener', will mimic basic behavior of one load balancer with one pool of application servers.  This will be used for naming server pool backends.  Application servers must only provide 1 host name and will join server pool backends using this name (e.g., www.mysite.com, api.mysite.com, default.mysite.com).",
+  :description => "Comma-separated list of host names for which the load balancer will answer website requests. First entry will be the default backend and will answer for all host names not listed here. A single entry of any name, e.g. 'default' or 'applistener', will mimic basic behavior of one load balancer with one pool of application servers. This will be used for naming server pool backends. Application servers must only provide 1 host name and will join server pool backends using this name (e.g., www.mysite.com, api.mysite.com, default.mysite.com).",
```
Comment (ekargard) [inline cookbooks/lb/metadata.rb:26]: After the end of sentences 2 spaces is the standard that is based on type setting and  manual type writers.  I know there are many arguments to change this.  Are we adopting a single space?

For 45 years 2 spaces has been burned into my brain.  Every book and printed article I read re-enforces that.

Having to scroll a line of text and look at the changes just to find out only spaces after periods where removed is tiresome.

My opinion is who cares.  Type how you are comfortable.  If it's not how you would type let it be.  Only make relevant  changes to the text.  Anything else leave be and don't waste time.

File: cookbooks/lb/metadata.rb (hunk 1 above)
Comment (ekargard) [inline cookbooks/lb/metadata.rb]: The formatting should have been left alone and all other cases of these mega lines should be converted into one line per entry. 

Look at this diff.  By merging it back into a single line it is impossible to read with out scrolling.

We need to be able to review all code within the text box that github forces code to be in.

File: cookbooks/lb/metadata.rb [hunk 8]
```diff
@@ -23,118 +23,79 @@
...
 
 attribute "lb/stats_uri",
   :display_name => "Status URI",
   :description => "The URI for the load balancer statistics report page (e.g., /haproxy-status). This page lists the current session, queued session, response error, health check error, server status, etc. for each load balancer group.",
   :required => "optional",
   :default => "/haproxy-status",
-  :recipes => [
-                'lb::setup_load_balancer'
-                ]
+  :recipes => ["lb::setup_load_balancer"]
```
Comment (ekargard) [inline cookbooks/lb/metadata.rb]: This is the standard we want.  Please revert the changes

File: cookbooks/lb/metadata.rb [hunk 9]
```diff
@@ -23,118 +23,79 @@
...
 attribute "lb/service/account_secret",
   :display_name => "Load Balance Service Secret",
   :description => "If using Rackspace Cloud Load Balancing, specify the Rackspace API key to use for authentication (e.g. cred:RACKSPACE_AUTH_KEY).",
-  :recipes => [
-                'lb::default',
-                'lb::do_attach_request',
-                'lb::do_detach_request',
-                ],
+  :recipes => ["lb::default", "lb::do_attach_request", "lb::do_detach_request"],
   :required => "optional"
```
Comment (ekargard) [inline cookbooks/lb/metadata.rb]: All of the multi-line recipes merged into a single line need to be reverted.  And all other instances of a single line like these need to be converted into multi-line entries.

Comment (ekargard) [inline cookbooks/lb/metadata.rb]: Also all recipe lists should be in the identical order.  If two similar inputs have the same list of recipes we need to be able to do a side by side comparison.
